doc/en/doctest.rst

```diff
@@ -128,20 +131,58 @@ itself:
     >>> get_unicode_greeting()  # doctest: +ALLOW_UNICODE
     'Hello'
 
-By default, pytest would report only the first failure for a given doctest.  If
+By default, pytest would report only the first failure for a given doctest. If
 you want to continue the test even when you have failures, do:
 
 .. code-block:: bash
 
     pytest --doctest-modules --doctest-continue-on-failure
 
 
-.. _`doctest_namespace`:
+Output format
+-------------
+
+You can change the diff output format on failure for your doctests
+by using one of standard doctest modules format in options
+(see :data:`python:doctest.REPORT_UDIFF`, :data:`python:doctest.REPORT_CDIFF`,
+:data:`python:doctest.REPORT_NDIFF`, :data:`python:doctest.REPORT_ONLY_FIRST_FAILURE`):
+
+.. code-block:: bash
+
+    pytest --doctest-modules --doctest-report none
+    pytest --doctest-modules --doctest-report udiff
+    pytest --doctest-modules --doctest-report cdiff
+    pytest --doctest-modules --doctest-report ndiff
+    pytest --doctest-modules --doctest-report only_first_failure
 
-The 'doctest_namespace' fixture
--------------------------------
 
-.. versionadded:: 3.0
+pytest-specific features
+------------------------
+
+Some features are provided to make writing doctests easier or with better integration with
+your existing test suite. Keep in mind however that by using those features you will make
+your doctests incompatible with the standard ``doctests`` module.
+
+Using fixtures
+^^^^^^^^^^^^^^
+
+It is possible to use fixtures using the ``getfixture`` helper:
+
+.. code-block:: text
+
+    # content of example.rst
+    >>> tmp = getfixture('tmpdir')
```

Comment:
Hah! I did this slightly differently in Sybil, not sure which way I prefer but I'd be interested in seeing the implementation and wiring for `getfixture`!

Reply:
Sure, here it is:

https://github.com/pytest-dev/pytest/blob/9c700d1fd5496c2f844f3dd20835ab204a822945/src/_pytest/doctest.py#L189-L197

Basically we inject `request.getfixturevalue` as `getfixture` into the namespace.